src/Microsoft.Extensions.CodeGeneration.Utils/DotNet/LibraryExportExtensions.cs

```diff
@@ -14,17 +14,28 @@ namespace Microsoft.Extensions.CodeGeneration.DotNet
 {
     public static class LibraryExportExtensions
     {
-        public static IEnumerable<MetadataReference> GetMetadataReferences(this LibraryExport export)
+        public static IEnumerable<MetadataReference> GetMetadataReferences(this LibraryExport export, bool throwOnError = true)
         {
             var references = new List<MetadataReference>();
             AssemblyMetadata assemblyMetadata;
             foreach (var lib in export.CompilationAssemblies)
             {
-                using (var stream = File.OpenRead(lib.ResolvedPath))
+                try
                 {
-                    var moduleMetadata = ModuleMetadata.CreateFromStream(stream, PEStreamOptions.PrefetchMetadata);
-                    assemblyMetadata = AssemblyMetadata.Create(moduleMetadata);
-                    references.Add(assemblyMetadata.GetReference());
+                    using (var stream = File.OpenRead(lib.ResolvedPath))
+                    {
+                        var moduleMetadata = ModuleMetadata.CreateFromStream(stream, PEStreamOptions.PrefetchMetadata);
+                        assemblyMetadata = AssemblyMetadata.Create(moduleMetadata);
+                        references.Add(assemblyMetadata.GetReference());
+                    }
+                }
+                catch (Exception ex) 
+                {
+                    if(throwOnError) 
+                    {
+                        throw ex;
+                    }
+                    continue;
```

Comment:
hm did you just list all exceptions from CreateFromStream and Create? I meant if there are some specific scenarios which cause a specific exception  that you aware of and expect then we should handle that . However if you want to protect from any exceptions to avoid throwing to user , then generic Exception should work as soon as we report it somehow differently.